opentracing-impl/src/main/java/io/opentracing/impl/AbstractActiveSpan.java

```diff
@@ -0,0 +1,158 @@
+package io.opentracing.impl;
+
+import io.opentracing.ActiveSpan;
+import io.opentracing.ActiveSpanSource;
+import io.opentracing.Span;
+import io.opentracing.SpanContext;
+
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * {@link AbstractActiveSpan} deals with {@link ActiveSpanSource}/{@link ActiveSpan}/{@link Continuation}
+ * reference counting while allowing subclasses to extend with bindings to ThreadLocals, etc.
+ */
+public abstract class AbstractActiveSpan implements ActiveSpan {
```

Comment:
:+1: Glad to see actual `public abstract` base classes instead of package-protected ones that force all implementations into the same package.